csp_tools/aws/nccl.sh

```diff
@@ -17,12 +17,39 @@
 #SBATCH --ntasks-per-node=8
 #SBATCH --time=00:20:00
 
-export NCCL_TOPO_FILE=/nccl/p4d-24xl-topo.xml
+# Load libraries
+export NCCL_TOPO_FILE="/nccl/topo.xml"
+export 
+LD_LIBRARY_PATH="/opt/nccl/build/lib:/usr/local/cuda/lib64:/opt/amazon/efa/lib64:/opt/amazon/openmpi/lib:/opt/aws-ofi-nccl/lib:$LD_LIBRARY_PATH"
+
+# EFA configurations
+export FI_PROVIDER=efa
+
+# NCCL configurations
+export NCCL_DEBUG=info
+export NCCL_PROTO=simple
+export NCCL_BUFFSIZE=33554432
+export NCCL_ALGO=ring
+export NCCL_DEBUG_SUBSYS=init,net,graph,env
+export NCCL_MIN_NCHANNELS=32
+
+# ompi conf
+export OPAL_PREFIX=/opt/amazon/openmpi
+export OMPI_MCA_btl=tcp,self
+export OMPI_MCA_pml=^cm
+export OMPI_MCA_btl_tcp_if_exclude=lo,docker0
+
+#pmix conf
+export PMIX_MCA_gds=^ds12
+export PMIX_MCA_btl=tcp,self
+export PMIX_MCA_pml=^cm
+export PMIX_MCA_btl_tcp_if_exclude=lo,docker0
 
 env | grep "SLURMD_NODENAME="
 env | grep "SLURM_NODELIST="
 
-srun --mpi=pmix --container-image=../../nemo_megatron_training.sqsh \
+module load openmpi
+srun --mpi=pmix --nodes=2 --tasks-per-node=8 --container-image=../../nemo_megatron_training.sqsh \
      --container-mounts="$PWD:/nccl" \
      bash -c "
-     /nccl/nccl-tests/build/all_reduce_perf -b 256M -e 8G -f 2 -c 1 -n 10"
+     /nccl/nccl-tests/build/all_reduce_perf -b 8 -e 16G -f 2 -g 1 -c 0 -n 100"
```

Comment:
can we keep the test `all_reduce_perf` args the same? this might affect what `cluster-validation.sh` marks as pass or fail.

Reply:
this still marks it as pass, but sure I can revert it

Reply:
done